lib/pharos/up_command.rb

```diff
@@ -152,42 +141,41 @@ def worker_hosts(config)
       config.hosts.select { |h| h.role == 'worker' }
     end
 
-    # @param master [Pharos::Configuration::Node]
-    # @param features [Pharos::Config]
-    def handle_masters(master, config)
-      log_host_header(master)
-      Phases::ConfigureHost.new(master).call
-      Phases::ConfigureMaster.new(master, config).call
-      Phases::ConfigureClient.new(master).call
-      Phases::ConfigureDNS.new(master, config).call
-      Phases::ConfigureNetwork.new(master, config.network).call
-      Phases::ConfigureMetrics.new(master).call
-      Phases::LabelNode.new(master, master).call
-      Phases::StoreClusterYAML.new(master, config_yaml.read(ENV.to_h)).call
+    def handle_phases(config)
+      all_hosts = config.hosts
+      master_hosts = master_hosts(config)
+      master_host = master_hosts[0]
+      worker_hosts = worker_hosts(config)
```

Comment:
This part is a bit of a mess, I think the `config` should be stored as an instance variable instead of getting passed around as a function argument in order to clean this up...